Sources/TuistKit/Services/GraphImplicitImportLintService.swift

```diff
@@ -0,0 +1,50 @@
+import Path
+import Tools
+import TuistCore
+import TuistGenerator
+import TuistLoader
+import XcodeGraph
+
+final class GraphImplicitImportLintService {
+    let targetScanner: TargetImportsScanning
+
+    init(targetScanner: TargetImportsScanning = TargetImportsScanner()) {
+        self.targetScanner = targetScanner
+    }
+
+    func lint(graphTraverser: GraphTraverser, config _: Config) async throws -> [LintingIssue] {
+        let allTargets = graphTraverser
+            .allTargets()
+
+        let allTargetNames = Set(allTargets.map(\.target.productName))
+
+        var implicitTargetImports: [Target: Set<String>] = [:]
+        for project in graphTraverser.projects.values {
+            let allTargets = project.targets.values
+
+            for target in allTargets {
+                let targetImports = Set(try await targetScanner.imports(for: target))
+                let targetTuistDeclaredDependencies = target.dependencies.compactMap {
+                    switch $0 {
+                    case let .target(name: targetName, _):
+                        return project.targets[targetName]?.productName
+                    case let .project(target: targetName, path: projectPath, _):
+                        return graphTraverser.projects[projectPath]?.targets[targetName]?.productName
+                    default:
+                        return nil
+                    }
+                }
+                let implicitImports = targetImports.intersection(allTargetNames).subtracting(targetTuistDeclaredDependencies)
```

Comment:
What happens if I import `X` where `X` is declared as a dependency with a pre-compiled XCFramework? Will that cause this to fail?

Reply:
I started to use allInternalTargets method here. Before that you could see imports which are marked as implicit, but they aren't defined in tuist manifest. Check it out [here](https://github.com/tuist/tuist/pull/6603)